subsys/net/ip/l2/ieee802154/ieee802154.c

```diff
@@ -291,8 +291,27 @@ static u16_t ieee802154_reserve(struct net_if *iface, void *data)
 	return ieee802154_compute_header_size(iface, (struct in6_addr *)data);
 }
 
+static int ieee802154_enable(struct net_if *iface, bool state)
```

Comment:
The name is a bit misleading as this can disable the radio too. What about calling this "ieee802154_set_state" or similar?

Reply:
This function is here to implement net_l2's enable() function. I won't give it a name that does not fit with this API. (Now if net_l2's enable() needs to be renamed, that's another issue).